common/src/main/java/org/apache/sedona/common/raster/RasterConstructors.java

```diff
@@ -96,10 +96,8 @@ public static GridCoverage2D makeEmptyRaster(int numBand, int widthInPixel, int
         // Create a new empty raster
         WritableRaster raster = RasterFactory.createBandedRaster(DataBuffer.TYPE_DOUBLE, widthInPixel, heightInPixel, numBand, null);
         MathTransform transform = new AffineTransform2D(scaleX, skewY, skewX, -scaleY, upperLeftX + scaleX / 2, upperLeftY - scaleY / 2);
-        GridGeometry2D gridGeometry = new GridGeometry2D(new GridEnvelope2D(0, 0, widthInPixel, heightInPixel), transform, crs);
-        ReferencedEnvelope referencedEnvelope = new ReferencedEnvelope(gridGeometry.getEnvelope2D());
         // Create a new coverage
         GridCoverageFactory gridCoverageFactory = CoverageFactoryFinder.getGridCoverageFactory(null);
-        return gridCoverageFactory.create("genericCoverage", raster, referencedEnvelope);
+        return gridCoverageFactory.create("genericCoverage", raster, crs, transform, null, null, null, null, null);
```

Comment:
We can specify the anchor to be `PixelInCell.CELL_CORNER` to follow the convention of GDAL and PostGIS. I have some code for doing this, you can take a look.

```java
        // Create a new empty raster
        int rasterDataType = DataBuffer.TYPE_DOUBLE;
        WritableRaster raster = RasterFactory.createBandedRaster(rasterDataType, widthInPixel, heightInPixel, numBand, null);
        MathTransform transform = new AffineTransform2D(scaleX, skewY, skewX, scaleY, upperLeftX, upperLeftY);
        GridGeometry2D gridGeometry = new GridGeometry2D(
                new GridEnvelope2D(0, 0, widthInPixel, heightInPixel),
                PixelInCell.CELL_CORNER,
                transform, crs, null);

        // Create a new coverage
        final ColorSpace cs = new BogusColorSpace(numBand);
        final int[] nBits = new int[numBand];
        Arrays.fill(nBits, DataBuffer.getDataTypeSize(rasterDataType));
        ColorModel colorModel =
                new ComponentColorModel(cs, nBits, false, true, Transparency.OPAQUE, rasterDataType);
        final RenderedImage image = new BufferedImage(colorModel, raster, false, null);
        GridCoverageFactory gridCoverageFactory = CoverageFactoryFinder.getGridCoverageFactory(null);
        return gridCoverageFactory.create("genericCoverage", image, gridGeometry, null, null, null);
```